go/vt/wrangler/vexec.go

```diff
@@ -0,0 +1,394 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wrangler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/olekukonko/tablewriter"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/concurrency"
+	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/topo"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/evalengine"
+)
+
+const (
+	vreplicationTableName = "_vt.vreplication"
+)
+
+type vexec struct {
+	ctx      context.Context
+	workflow string
+	keyspace string
+	query    string
+
+	wr *Wrangler
+
+	plan    *vexecPlan
+	masters []*topo.TabletInfo
+}
+
+func newVExec(ctx context.Context, workflow, keyspace, query string, wr *Wrangler) *vexec {
+	return &vexec{
+		ctx:      ctx,
+		workflow: workflow,
+		keyspace: keyspace,
+		query:    query,
+		wr:       wr,
+	}
+}
+
+// VExec executes queries on _vt.vreplication on all masters in the target keyspace of the workflow
+func (wr *Wrangler) VExec(ctx context.Context, workflow, keyspace, query string, dryRun bool) (map[*topo.TabletInfo]*sqltypes.Result, error) {
+	results, err := wr._vexec(ctx, workflow, keyspace, query, dryRun)
+	retResults := make(map[*topo.TabletInfo]*sqltypes.Result)
+	for tablet, result := range results {
+		retResults[tablet] = sqltypes.Proto3ToResult(result)
+	}
+	return retResults, err
+}
+
+func (wr *Wrangler) _vexec(ctx context.Context, workflow, keyspace, query string, dryRun bool) (map[*topo.TabletInfo]*querypb.QueryResult, error) {
```

Comment:
Is there a reason for the underscore prefix? In go, there's no special meaning for it. May be better to just name it `vexec`.